### DISCUSSION TITLE
blog/codeium-analysis-4-2024

### DISCUSSION THREAD

**Discussion by giscus** [zackproser/portfolio#444] [Announcements]
# blog/codeium-analysis-4-2024

It's been about a year since I last looked at Codeium - what has changed?

https://zackproser.com/blog/codeium-analysis-4-2024

**Comment by zackproser** [comment]
Codeium live is pretty useful, especially if you've been bitten by ChatGPT or Claude giving you Next.js code that is still using the /pages router!

**Comment by vtarsh** [comment]
In my opinion, the implementation is still quite weak, even with a paid PRO period, comparable to Cursor.
I share my experience using both IDEs based on the Claude 3.5 Sonnet model

Cons:
When you directly specify a folder for the context, it selects only partial files and does not see everything else, even when you specifically ask it to analyze all the files in the folder, so the context is very broken.
Autocomplete is very poorly implemented compared to Cursor, and also offers very stupid autocompletes, for example, deleting working pieces of code that the question had nothing to do with.
It often hallucinates, supplementing the code with non-existent pieces of code that have nothing to do with logic at all and creates incorrect imports or non-existent ones at all.
There is no or I did not find a setting specifically for Codeium AI, where you can set a prompt instruction, for example, answer in a certain language, etc. This is strange, because when installing Codeium AI as a separate extension in VS Code, these settings are present.
There is no option to select beta features, such as greater context, like in Cursor in 200k, using Claude. I am willing to pay much more than $10+ for such additional features.

Pros:
Has very good potential due to the use of searching for specific files through imports for better context, but this needs improvement for now.
No bugs, like in Cursor, with loss of context when pasting parts of code directly into the chat.

Summary:
So far, a very raw product that needs significant improvements, but has good potential due to specific search by project. Overall, I would like to say that I like it, but there are still many minuses. I would like to have beta features such as greater context or analysis of all files in a specified directory, not just those that it chooses itself.
I will follow the product from time to time and wish the developers success.
